src/pages/how-does-react-tell-a-class-from-a-function/index.zh-hant.md

```diff
@@ -0,0 +1,553 @@
+---
+title: React 是如何從 Function 中區別出 Class？
+date: '2018-12-02'
+spoiler: 我們談論關於類別、new、instanceof、原型鏈（prototype chains）、和 API 設計。
+---
+
+考慮一下這個被定義為函式的 `Greeting` 元件：
+
+```jsx
+function Greeting() {
+  return <p>Hello</p>;
+}
+```
+
+React 也支援把它定義為類別：
+
+```jsx
+class Greeting extends React.Component {
+  render() {
+    return <p>Hello</p>;
+  }
+}
+```
+
+（直到 [近期](https://reactjs.org/docs/hooks-intro.html) 前，這曾是唯一能使用像是 state 功能的方法。）
+
+當你想要繪製一個 `<Greeting />` 時，你不需要煩惱它是如何被定義的：
+
+```jsx
+// 類別或函式 — 隨意。
+<Greeting />
+```
+
+但是 *React 本身* 在乎它的不同！
+
+如果 `Greeting` 是一個函式, React 需要呼叫它：
+
+```jsx
+// 你的程式碼
+function Greeting() {
+  return <p>Hello</p>;
+}
+
+// React 內
+const result = Greeting(props); // <p>Hello</p>
+```
+
+但如果 `Greeting` 是一個類別，React 需要用 `new` 運算符實現出它的實例，*再* 呼叫剛才創建的實例中的 `render` 方法：
+
+```jsx
+// 你的程式碼
+class Greeting extends React.Component {
+  render() {
+    return <p>Hello</p>;
+  }
+}
+
+// React 內
+const instance = new Greeting(props); // Greeting {}
+const result = instance.render(); // <p>Hello</p>
+```
+
+在上述兩種情境中，React 的目的是拿到繪製過的節點（在這例子中，`<p>Hello</p>`），但是確切的步驟取決於 `Greeting` 是如何被定義的。
+
+**所以 React 是如何知道這個東西是類別還是函式？**
+
+正如我 [上一篇文章](/zh-hant/why-do-we-write-super-props/)，**在 React 中，你 *不需要* 知道這個也能具有成效，**我也是多年以來都不知道。請不要把它當成一道面試問題，事實上，這篇文章相較於涉及 React，更多在於涉及 JavaScript。
```

Comment:
```suggestion
正如我 [上一篇文章](/zh-hant/why-do-we-write-super-props/)所說的，**你 *不需要* 知道這個也能夠有效運用 React，**我多年以來也都不知道這兩者的差別。請不要把這變成一道面試問題。事實上，這篇文章與其說是討論 React 的運作，更多的是討論 JavaScript 如何運作。
```

Reply:
Maybe comma?
```diff
- 我多年以來也都不知道這兩者的差別。請不要把這變成一道面試題。
+ 我多年以來也都不知道這兩者的差別，請不要把這變成一道面試題。
```

Reply:
@chloewlin

Reply:
@qas612820704  The original text did not have a comma. They are two separated sentence.

Reply:
Okay. I just think that the comma is more smooth in Chinese.